### DISCUSSION TITLE
Remix on gh-pages?

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [remix-run/remix#2127] [Q&A] [answered]
Is there a way to use remix on gh-pages?

I have tried googling this and what not, but I haven't found anything that talks about this.

**Comment by sergiodxa** [comment, marked as answer]
Remix has a server-side part that GitHub Pages will not be able to run, so it will not be possible to run Remix on a pure static file hosting.

You could however, use Remix to build an API and fetch it from a SPA on GitHub Pages but you will lose most of the benefits of using Remix, at that point it's not different than CRA + any other API server.

**Reply by JacobWeisenburger** [reply]
That's what I thought, but I just wanted to know for sure. Thanks for the quick answer.

**Reply by brookslybrand** [reply]
Just adding to this since the answer is now outdated but this issue comes up on Google Searches still.

You can now do this with [SPA Mode](https://remix.run/docs/en/main/future/spa-mode). [Here's a demo](https://github.com/brookslybrand/remix-gh-pages)

**Comment by xeoneux** [comment]
Hi, can Remix now be deployed on GH Pages with the latest SPA build?

**Reply by sergiodxa** [reply]
Yes, it should be possible, but because GH Pages doesn't redirect every route to the index.html if the user is on any route that's not `/` and reload the tab, it will fails to load the HTML and start the app again.

**Reply by xeoneux** [reply]
Is there any plan for exporting individual pages like Next.js does?

**Reply by sergiodxa** [reply]
There has been some discussions about adding an SSG mode, you could also duplicate the index.html to match every route you need to handle after you run the build script.

**Comment by aryaan22a** [comment]
![IMG_20251029_204537_474.jpg](https://github.com/user-attachments/assets/2bb2894b-dc1a-4d64-97ab-301610a304bb)

**Reply by aryaan22a** [reply]
Usdt withdrawal not calling

**Reply by sergiodxa** [reply]
This is not the repo of Remix IDE, and this discussion has nothing to do with your particular issue anyway
